File: dahui/2023_10/1013/PGS_기사단원의무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 1;
        
        for(int i=2; i<=number; i++){ //모든 기사들의 약수 개수 구하기
            int nightPower = 0;
            int max = i;
            for(int j=1; j<max; j++){ //한 수의 약수 개수 구하기
                if(i%j == 0){
                    if((i/j) == j ) {
                        nightPower += 1;
                        max = i/j;
                    }else {
                       max = i/j;
                        nightPower += 2; 
                    }
                }
            }
            if(nightPower > limit) answer += power;
            else answer += nightPower;
        }
        return answer;
    }
}

File: woojae/2023_12/1227/PGS_기사단원의 무기.py
def solution(number, limit, power):
    answer = [1]
    for i in range(2, number+1):
        temp = []
        for j in range(1, int(i**0.5)+1):  # 시간 단축
            if i % j == 0:  # i가 j로 나눠 떨어지면 j는 i의 약수
                temp.append(j)
                temp.append(i // j)  # j x (i // j) = 1 x 4, 2 x 2의 형태로 약수는 1, 2, 4가 됨
        answer.append(len(set(temp)))  # 중복되는 수를 하나의 약수로 표현
    return sum([i if i <= limit else power for i in answer])  # 조건에 맞춰 필터링 후 합산
